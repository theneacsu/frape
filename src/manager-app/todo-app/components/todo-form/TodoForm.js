import React, { Component } from 'react'
import styles from './TodoForm.module.css'

class TodoForm extends Component {
  state = {
    todoText: this.props.todo ? this.props.todo.text : '',
    completed: this.props.todo ? this.props.todo.completed : false,
    error: undefined
  }

  onInputChange = e => {
    const todoText = e.target.value
    this.setState(() => ({ todoText }))
  }

  onSubmitHandler = e => {
    e.preventDefault()

    if (this.state.todoText.trim().length > 2) {
      this.props.onSubmit({
        text: this.state.todoText,
        completed: this.state.completed
      })
      this.setState(() => ({ error: undefined, todoText: '' }))
    } else {
      const error = 'The todo must have at least 3 characters.'
      this.setState(() => ({ error }))
    }
  }

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmitHandler} className={styles.form}>
          <input
            className={['input', styles.input].join(' ')}
            type="text"
            placeholder={this.props.placeholder}
            value={this.state.todoText}
            onChange={this.onInputChange}
          />
          <button className={styles.button}>{this.props.buttonText}</button>
        </form>
      </div>
    )
  }
}

export default TodoForm
