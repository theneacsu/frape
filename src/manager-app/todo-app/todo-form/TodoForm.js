import React from 'react'

const TodoForm = props => (
  <div>
    <form>
      <input type="text" placeholder={props.placeholder} />
      <button>{props.buttonText}</button>
    </form>
  </div>
)

export default TodoForm
