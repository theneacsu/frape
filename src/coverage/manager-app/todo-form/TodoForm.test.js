import React from 'react'
import { shallow } from 'enzyme'
import TodoForm from '../../../manager-app/todo-app/todo-form/TodoForm'

test('Should render correctly the TodoForm component for adding a todo', () => {
  const wrapper = shallow(
    <TodoForm placeholder="ex: Go for a run" buttonText="Add todo" />
  )
  expect(wrapper).toMatchSnapshot()
})

test('Should render correctly the TodoForm component for editing a todo', () => {
  const wrapper = shallow(<TodoForm buttonText="Save" />)
  expect(wrapper).toMatchSnapshot()
})
