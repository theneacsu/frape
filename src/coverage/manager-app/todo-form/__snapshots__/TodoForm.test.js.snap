// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the TodoForm component for adding a todo 1`] = `
<div>
  <form>
    <input
      placeholder="ex: Go for a run"
      type="text"
    />
    <button>
      Add todo
    </button>
  </form>
</div>
`;

exports[`Should render correctly the TodoForm component for editing a todo 1`] = `
<div>
  <form>
    <input
      type="text"
    />
    <button>
      Save
    </button>
  </form>
</div>
`;
